### PR TITLE
Updated the faucet to use `fuel-core` 0.10.1

### PR DESCRIPTION
**Pull request by xgreenx**
Prepared the faucet for the upcoming change in `fuels-rs` https://github.com/FuelLabs/fuels-rs/pull/546

Closes https://github.com/FuelLabs/faucet/issues/11

### CODE REVIEW

**Review by luizstacio — DISMISSED** [review]

**Comment by luizstacio** [comment]
@xgreenx it's possible to receive from the faucet via API by sending a bech32 address?

**Review by Voxelot — DISMISSED** [review]

**Comment by Voxelot** [comment]
@luizstacio assuming fuel-rs provides a way to convert a Bech32 encoded string into a fuel_types::Address, then it should be possible. Should we accept both hex based and bech32 representations, or is it fine to only support bech32?

**Comment by luizstacio** [comment]
> @luizstacio assuming fuel-rs provides a way to convert a Bech32 encoded string into a fuel_types::Address, then it should be possible. Should we accept both hex based and bech32 address representations, or is it fine to only support bech32?

Would be good to support both

File: src/lib.rs [hunk 1]
```diff
@@ -153,17 +153,20 @@ fn init_logger(config: &Config) {
...
         // use machine parseable structured logs
         sub
             // disable terminal colors
             .with_ansi(false)
             // use json
             .json()
-            .init();
+            .try_init()
+    };
+    if let Err(err) = inited {
```
Comment (Voxelot) [inline src/lib.rs:168]: If we can't init the logger we shouldn't start the service imo

Comment (xgreenx) [reply src/lib.rs:168]: In unit tests we can't init it two timtes=(

Comment (Voxelot) [reply src/lib.rs:168]: Hmm we should move the logger init to a higher level fn than `start_service`, and only call it from main then.

**Review by Voxelot — APPROVED** [review]

**Review by luizstacio — APPROVED** [review]

**Pull request merged**